pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go

```diff
@@ -1094,22 +1130,112 @@ func (p *DynamicPolicy) getContainerRequestedCores(allocationInfo *state.Allocat
 		return 0
 	}
 
-	if allocationInfo.RequestQuantity == 0 {
-		if p.metaServer == nil {
-			general.Errorf("got nil metaServer")
-			return 0
+	if p.metaServer == nil {
+		general.Errorf("got nil metaServer")
+		return allocationInfo.RequestQuantity
+	}
+
+	container, err := p.metaServer.GetContainerSpec(allocationInfo.PodUid, allocationInfo.ContainerName)
+	if err != nil || container == nil {
+		general.Errorf("get container failed with error: %v", err)
+		return allocationInfo.RequestQuantity
+	}
+
+	cpuQuantity := native.CPUQuantityGetter()(container.Resources.Requests)
+	metaValue := general.MaxFloat64(float64(cpuQuantity.MilliValue())/1000.0, 0)
+
+	// optimize this logic someday:
+	//	only for refresh cpu request for old pod with cpu ceil and old VPA pods.
+	if state.CheckShared(allocationInfo) {
+		// if the pod is created after support vpa resizing, we don't need to check the pod request from podWatcher anymore.
+		if util.PodCreatedAfterSupportVPAResizing(allocationInfo.Annotations) {
+			return allocationInfo.RequestQuantity
+		}
+		if state.CheckNUMABinding(allocationInfo) {
+			if metaValue < allocationInfo.RequestQuantity {
+				general.Infof("[snb] get cpu request quantity: (%.3f->%.3f) for pod: %s/%s container: %s from podWatcher",
+					allocationInfo.RequestQuantity, metaValue, allocationInfo.PodNamespace, allocationInfo.PodName, allocationInfo.ContainerName)
+				return metaValue
+			}
+		} else {
+			if metaValue != allocationInfo.RequestQuantity {
+				general.Infof("[share] get cpu request quantity: (%.3f->%.3f) for pod: %s/%s container: %s from podWatcher",
+					allocationInfo.RequestQuantity, metaValue, allocationInfo.PodNamespace, allocationInfo.PodName, allocationInfo.ContainerName)
+				return metaValue
+			}
 		}
+	} else if allocationInfo.RequestQuantity == 0 {
+		general.Infof("[other] get cpu request quantity: (%.3f->%.3f) for pod: %s/%s container: %s from podWatcher",
+			allocationInfo.RequestQuantity, metaValue, allocationInfo.PodNamespace, allocationInfo.PodName, allocationInfo.ContainerName)
+		return metaValue
+	}
+
+	return allocationInfo.RequestQuantity
+}
 
-		container, err := p.metaServer.GetContainerSpec(allocationInfo.PodUid, allocationInfo.ContainerName)
-		if err != nil || container == nil {
-			general.Errorf("get container failed with error: %v", err)
-			return 0
+func (p *DynamicPolicy) checkNormalShareCoresCpuResource(req *pluginapi.ResourceRequest) (bool, error) {
+	_, reqFloat64, err := util.GetQuantityFromResourceReq(req)
```

Comment:
why not using util.GetPodAggregatedRequestResource(req) in normal shared_cores candidate

Reply:
QRM plugin records per container allocation for share cores pod containers, we just use the aggregated request in GetTopologyHints for share cores pod.